### PR TITLE
Bug/subdomain name crash fix

### PR DESCRIPTION
**Pull request by cfeilen**
Checks to make sure that _serviceHandlers contains the key before we attempt to access it via the indexer operation.
Fixes CR/LF in my change, and one other place.

### CODE REVIEW

**Comment by cfeilen** [comment]
Fixes #26

File: src/Tmds.MDns/NetworkInterfaceHandler.cs [hunk 1]
```diff
@@ -453,6 +453,11 @@ private void RemoveService(Name name)
             if (service != null)
             {
                 _serviceInfos.Remove(name);
+                if (!_serviceHandlers.ContainsKey(name.SubName(1)))
+                {
+                    return;
```
Comment (tmds) [inline src/Tmds.MDns/NetworkInterfaceHandler.cs]: The service should not have been in _serviceInfos if there is no _serviceHandler for that name.
How did the service get in the _serviceInfos?

File: src/Tmds.MDns/NetworkInterfaceHandler.cs [hunk 2]
```diff
@@ -488,6 +493,11 @@ private void HandlePacketServiceInfos()
                     {
                         service = packetService;
                         _serviceInfos.Add(packetName, service);
+                        if (!_serviceHandlers.ContainsKey(packetName.SubName(1)))
+                        {
+                            continue;
```
Comment (tmds) [inline src/Tmds.MDns/NetworkInterfaceHandler.cs]: Similar, the _packetServiceInfos should only contain services for which we have a handler.
This service should not have made it into the _packetServiceInfos.

**Comment by tmds** [comment]
If you ever have some time to look into the issue further, I'd definitely take a fix.

**Pull request closed without merging**